### PR TITLE
Refactor specs

### PR DESCRIPTION
**Pull request by FriedSock**
Just a reorganisation and refactor of the feature specs, we will still need to add in some missing tests later, but it is easier to spot what is missing now that the descriptions and structure is better.

### CODE REVIEW

File: spec/features/cco/dashboard_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      end
+
+      within "#defence_request_#{dr_open.id}" do
+        expect(page).to have_content(dr_open.solicitor_name)
+      end
+
+      dr_created.update(solicitor_name: "New Solicitor")
+      dr_open.update(solicitor_name: "New Solicitor2")
+      #Wait for dashboard refresh + 1
+      sleep(4)
```
Comment (dp6ai) [inline spec/features/cco/dashboard_spec.rb]: can we override the wait time to reduce...

File: spec/features/cco/defence_request_management_spec.rb [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+          expect(page).not_to have_link("Edit")
+        end
+      end
+
+      #TODO - make better
+      specify "can open the request" do
+        visit root_path
+        within "#defence_request_#{unopened_dr.id}" do
+          click_button "Open"
+        end
```
Comment (dp6ai) [inline spec/features/cco/defence_request_management_spec.rb]: expectation? todo: make it do something

Comment (FriedSock) [inline spec/features/cco/defence_request_management_spec.rb]: this is a reorganization, not refactor, don't shoot the messenger

File: spec/features/cco/defence_request_management_spec.rb [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+            click_button "Accepted"
+          end
+          within ".accepted-defence-request" do
+            expect(page).to have_content(opened_dr.solicitor_name)
+          end
+        end
+      end
+
+      context "that have the \"duty\" solicitor type" do
+        let!(:duty_solicitor_dr) { create(:defence_request, :duty_solicitor, :opened, cco: User.first) }
```
Comment (dp6ai) [inline spec/features/cco/defence_request_management_spec.rb]: User.first isn't great

File: spec/features/cco/defence_request_management_spec.rb [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+      within ".accepted-defence-request" do
+        click_link "Show"
+      end
+      expect(page).to have_content("1 January 2001 - 01:01")
+    end
+  end
+end
+
+
+
```
Comment (dp6ai) [inline spec/features/cco/defence_request_management_spec.rb]: Loads of lines...

File: spec/features/cso/defence_request_management_spec.rb [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+          fill_in "Age", with: "39"
+          fill_in "defence_request_date_of_birth_year", with: "1976"
+          fill_in "defence_request_date_of_birth_month", with: "01"
+          fill_in "defence_request_date_of_birth_day", with: "01"
+          choose "No"
+        end
+        fill_in "Comments", with: "This is a very bad man. Send him down..."
+        click_button "Create Defence Request"
+      end
+      an_audit_should_exist_for_the_defence_request_creation
```
Comment (dp6ai) [inline spec/features/cso/defence_request_management_spec.rb]: I think we maybe want 'audit specs'

File: spec/features/solicitor/dashboard_spec.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        within ".accepted-defence-request" do
+          expect(page).to have_content(solicitor_dr.detainee_name)
+          expect(page).to_not have_content(solicitor_dr_not_accepted.detainee_name)
+        end
+        expect(page).to_not have_selector(".created-defence-request")
+        expect(page).to_not have_selector(".open-defence-request")
+
+        sleep(4)
+        wait_for_ajax
```
Comment (dp6ai) [inline spec/features/solicitor/dashboard_spec.rb]: see above comment

**Comment by dp6ai** [comment]
Looks good. I'll merge and you can add the changes later

**Pull request merged**